src/java_bytecode/generate_java_generic_type.cpp

```diff
@@ -358,13 +337,17 @@ irep_idt generate_java_generic_typet::build_generic_name(
         existing_generic_type.generic_type_arguments().end()) ==
         static_cast<int>(generic_original_class.generic_types().size()),
       "All generic types must be concretised");
-    auto generic_type_p = generic_original_class.generic_types().begin();
 
-    generic_name_buffer << build_generic_name_for_class_arguments(
-      generic_argument_p,
-      generic_type_p,
-      generic_original_class.generic_types().end(),
-      original_class_name);
+    generic_arguments_name_buildert build_generic_arguments(
+      build_argument_name);
+
+    while(generic_argument_p !=
```

Comment:
Unfortunately not. `generic_argument_p` is an iterator over the concrete arguments we want to add, including the ones for outer classes. So by the time the execution reaches this point, the iterator may not be at the beginning of the list but rather somewhere in the middle because some of the arguments have already been used in the loop for implicitly generic case above. There is an invariant few lines above that checks that we still have enough arguments at this point to add to the name.